runelite-client/src/main/java/net/runelite/client/plugins/gpu/GpuPlugin.java

```diff
@@ -1051,6 +1063,23 @@ public void draw()
 
 			gl.glUseProgram(glProgram);
 
+			int distance = drawDistance * Perspective.LOCAL_TILE_SIZE;
+			gl.glUniform1i(uniUseFog, config.enableFog() ? 1 : 0);
+			gl.glUniform4f(uniFogColor, config.fogColor().getRed() / 255f, config.fogColor().getGreen() / 255f, config.fogColor().getBlue() / 255f, 1f);
+			gl.glUniform1i(uniFogDepth, config.fogDepth() * Perspective.LOCAL_TILE_SIZE * Math.min(SCENE_TILES / 2, drawDistance) / 100);
+
+			// currently this doesn't care if the camera is in the fog
+			int fogWest = client.getCameraX2() - distance;
+			int fogEast = client.getCameraX2() + distance - Perspective.LOCAL_TILE_SIZE;
+			int fogSouth = client.getCameraZ2() - distance;
+			int fogNorth = client.getCameraZ2() + distance - Perspective.LOCAL_TILE_SIZE;
+			gl.glUniform4i(uniSceneBounds,
```

Comment:
Can all of this + the calculations be moved to the shader?

Reply:
Possibly. I'm not sure if that's necessary, though, since this only needs to be calculated once per frame.

Reply:
I think it should be tried.